libraries/AP_BoardConfig/AP_BoardConfig_CAN.cpp

```diff
@@ -0,0 +1,173 @@
+ /*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+ *   AP_BoardConfig_CAN - board specific configuration for CAN interface
+ */
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Common/AP_Common.h>
+#include "AP_BoardConfig_CAN.h"
+
+#if CONFIG_HAL_BOARD == HAL_BOARD_PX4 || CONFIG_HAL_BOARD == HAL_BOARD_VRBRAIN
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+
+#if HAL_WITH_UAVCAN
```

Comment:
#if CONFIG_HAL_BOARD == HAL_BOARD_PX4 is located little above that.
HAL_WITH_UAVCAN should not cover the whole file. AP_Param::GroupInfo should be declared.